### DISCUSSION TITLE
CSS minifying process removing percent character from CSS.

### DISCUSSION THREAD

**Discussion by ilkkag** [vercel/next.js#62823] [Help]
### Summary

So, I am using the new @property css rule (https://developer.mozilla.org/en-US/docs/Web/CSS/@property) to animate some gradient masks. The issue is that CSS minifying process replaces the value of 0% with 0 and that causes the animation to fail as the value is expected to be percentage, not an integer. It is working correctly in local dev environment but when deployed to Vercel, no luck. Is there any way to fix this problem? I am using the latest version of Next.js.

### Additional information

```js
Chrome gives this error message: "An @property rule is ignored because it contains an invalid property or is missing a required one"

My code that works locally:
@property --c {
  syntax: "<percentage>";
  inherits: false;
  initial-value: 0%;
}
```

**Comment by ilkkag** [comment]
No one?

**Comment by icyJoseph** [comment]
Hi,

This can be one of the PostCSS plugins used, but I am not sure which one.

It looks a lot like this other issue: https://github.com/vercel/next.js/issues/62002#issuecomment-1941988417

Most definitely not the actual same issue, but it is a start I guess.

What's happening is that when you do `next build`, the CSS is compressed, minified and uglified as much as possible to save bits of data. 

So, we have to figure out which step of that process is removing the unit.

**Reply by icyJoseph** [reply]
I wonder if you could set `optimizeCss` to false in your next.config.js ~ I am not sure if its false by default, I think it is, but you might as well try. 

We could also look into writing a small webpack config that strips the cssnano minifier.

**Reply by ilkkag** [reply]
Hey, thanks for your replies! I tried setting optimizeCss to false, no luck there. I found a "hack" around this for now, by altering the type to length and value to <XX>rem but yes, something that needs looking into in the near future.

**Reply by icyJoseph** [reply]
There's a way to turn off minify altogether. I'd want to see if that also fixes your issue, so that we can pinpoint the origin of it.

Would you mind creating a webpack key on the next.config file, and then target config.optimization.minimize to false. I think there's also another property that can be set to an empty array to disable all minifiers.

**Reply by ilkkag** [reply]
Sure, that config change fixed it.

**Reply by icyJoseph** [reply]
Alright then it is most likely cssnano that's doing something weird. 

Could you please try to create a reproduction example at Stackblitz? Just type next.new on your URL search bar, and enter. It'll take you to a Stackblitz playground ready with Next.js.

Do the smallest change necessary to see the error, save the project and paste back the link. I'll make sure to use the link with `next build`, to narrow things further.

**Reply by ilkkag** [reply]
Sure thing, here it is: https://stackblitz.com/edit/nextjs-dbqjyf?file=app%2Fglobals.css

**Reply by icyJoseph** [reply]
Alright, I got something.

```css
@property --test {
  syntax: "<percentage>";
  inherits: false;
  initial-value: 0%;
}

div {
  width: var(--test);
}
```

If we take that through `CSSNano`, it outputs:

```css
@property --test{syntax:"<percentage>";inherits:false;initial-value:0}div{width:var(--test)}
```

To run this locally:

```shell
yarn init -y && yarn add postcss-cli postcss cssnano
```

and then create `postcss.config.js` with these contents:

```js
module.exports = {
    plugins: [
        require('cssnano')({
            preset: 'default',
        }),
    ],
};
```

Then create `input.css` paste your code in and run `npx postcss input.css`, and it'll print the faulty code.

It might still be `PostCSS` though... gonna have to test that as well

**Reply by rstar2** [reply]
Same problem exists in Nextjs 15.2 (not really Next.js fault but still it's not fixed in postCSS or what's so ever)
